spec: tighten Collection items to post URIs only + document Collection in README (#122)
pubky/pubky-app-specs: README.md

}
```

**Note on `kind = collection`:**

Collection posts use a typed JSON envelope as their `content`. The envelope shape is:

```json
{
  "name": "AI papers",
  "description": "Best stuff",
  "items": [
    "pubky://userA/pub/pubky.app/posts/0034A0X7NJ52A",
    "pubky://userB/pub/pubky.app/posts/0034A0X7NJ52B"
  ]
}
```

- `name` — required, 1–100 unicode scalars, non-whitespace-only.
- `description` — optional, max 500 scalars.

> - `items` — ordered list of URIs (≤ 100, each ≤ 300 chars, protocols in `[pubky, http, https]`).

Author: Good catch, fixed in 2fb6929. Updated README to state items must resolve to `Resource::Post` (form: `pubky://<pubky-id>/pub/pubky.app/posts/<post-id>`), and that file / profile / external URIs are rejected.

> - `items`: ordered list of pubky.app post URIs (max 100, each at most 300 chars). Each URI must resolve to a `Resource::Post` (form: `pubky://<pubky-id>/pub/pubky.app/posts/<post-id>`); all other URI types are rejected.

Reviewer: Remove reference to upper limit of 300 chars:

```suggestion
- `items`: ordered list of pubky.app post URIs (max 100). Each URI must resolve to a `Resource::Post` (form: `pubky://<pubky-id>/pub/pubky.app/posts/<post-id>`); all other URI types are rejected.
```

Reviewer: This reference also has to be removed from

https://github.com/pubky/pubky-app-specs/blob/c817c288d8fdbe915f5850e7a816852dadc9f318/src/limits.rs#L62

Author: Removing this reference from the comment made me thing the max length cap of 300 is actually dead code. I verified and actually there were still two ways `Url::parse` bypasses: userinfo padding (`pubky://JUNK@<pid>/...` strips via `host_str()`) and `..` path collapse. I rewote `validate_collection_item_uri` without `Url::parse` (`strip_prefix` + `split_once` + delegate to `PubkyId::try_from` and `validate_crockford_id`) closes both structurally; the length cap now  truly is dead code and I removed fully, fixed the comments as your suggestions. Two new regression tests to pin the bypasses 4093c109d18e13c0d39f67d27d562d7871add0c3

> - `items`: ordered list of pubky.app post URIs (max 100, each at most 300 chars). Each URI must resolve to a `Resource::Post` (form: `pubky://<pubky-id>/pub/pubky.app/posts/<post-id>`); file, profile, and external URIs are rejected.

Reviewer: ```suggestion
- `items`: ordered list of pubky.app post URIs (max 100, each at most 300 chars). Each URI must resolve to a `Resource::Post` (form: `pubky://<pubky-id>/pub/pubky.app/posts/<post-id>`); all other URI types are rejected.
```
